internal/controller/controllers/local_cluster_import_controller.go

```
@@ -0,0 +1,517 @@
+		return ctrl.Result{}, err
+	}
+
+	if instance.GetDeletionTimestamp().IsZero() {
+		r.log.Infof("Zero deletion timestamp")
+		// Check to see if the ASC has our finalizer, if so, we must add it
+		if !controllerutil.ContainsFinalizer(instance, agentServiceConfigLocalClusterImportFinalizerName) {
+			controllerutil.AddFinalizer(instance, agentServiceConfigLocalClusterImportFinalizerName)
+			if err := r.client.Update(ctx, instance); err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "failed to add finalizer to AgentServiceConfig")
+			}
+		}
+	} else {
+		r.log.Infof("Non zero deletion timestamp")
+		err := r.deleteLocalClusterImportCRs(ctx)
+		if err != nil {
+			r.log.WithError(err).Error("failed to clean up local cluster CRs")
+			return ctrl.Result{}, err
+		}
+		controllerutil.RemoveFinalizer(instance, agentServiceConfigLocalClusterImportFinalizerName)
+		if err := r.client.Update(ctx, instance); err != nil {
+			r.log.WithError(err).Error("failed to remove finalizer from AgentServiceConfig")
+			return ctrl.Result{}, err
+		}
+		// Stop reconciliation as the item is being deleted
+		return ctrl.Result{}, nil
+	}
+
+	hasManagedCluster, err := r.hasLocalManagedCluster(ctx)
+	if err != nil {
+		err = r.setReconciliationStatus(ctx, instance, false, aiv1beta1.ReasonUnableToDetermineLocalClusterManagedStatus, err.Error())
+		if err != nil {
+			r.log.Error("Unable to set reconciliation status of LocalClusterImport on AgentServiceConfig")
+		}
+		return ctrl.Result{}, errors.Wrap(err, "error while attempting to determine presence of managed cluster")
+	}
+	if hasManagedCluster {
+		if err = r.importLocalCluster(ctx, instance); err != nil {
+			return ctrl.Result{}, errors.Wrap(err, "failed to create managed cluster CRs")
+		}
+		err = r.setReconciliationStatus(ctx, instance, true, aiv1beta1.ReasonLocalClusterManaged, "")
+		if err != nil {
+			r.log.Error("Unable to set reconciliation status of LocalClusterImport on AgentServiceConfig")
```

I think we should return the error here so that we re-reconcile.